application/common/dos.py

```diff
@@ -125,6 +130,35 @@ def get_matching_dos_services(odscode: str) -> List[DoSService]:
     return services
 
 
+def build_select_statement():
```

Comment:
This seems like a really messy resolution to the problem it tries to fix, I think it may be better just to write out the db columns raw in the query string. I'm going to change the way the DoSService object creates fields in a newer branch so dont feel the need to change it now.